File: src/mainApp/Platform.java
package mainApp;

import java.awt.Graphics2D;
import java.awt.image.BufferedImage;
import java.io.File;
import java.io.IOException;

import javax.imageio.ImageIO;

/**
 * Class: Platform
 * @author Team 303
 * <br>Purpose: Class for a stationary platform in game.
 * <br>Restrictions: None
 */
public class Platform extends GameObject {

	private double WIDTH;
	private double HEIGHT;
	private boolean isLava;
	private boolean isIce;
	private boolean isSlime;
	private boolean isCool;
	private String name;

	
	public Platform(double xCord,double yCord, int type) {
		super(xCord, yCord);
		WIDTH = this.getWidth();
		HEIGHT = this.getHeight();
		this.isLava = false;
		this.isIce = false;
		this.isSlime = false;
		this.isCool = false;
		this.setName(type);
		if(isLava) {
			HEIGHT = HEIGHT*.6;
			this.setHeight(HEIGHT);
		}
	}
	
	public void setName(int a) { 
		switch (a) {
		case 0:
			this.name = "Truss";
			break;
		case 1:
			this.name = "Lava";
			this.isLava = true;
			break;
		case 2:
			this.name = "Ice";
			this.isIce = true;
			break;
		case 3:
			this.name = "Slime";
			this.isSlime = true;
			break;
		case 4:
			this.name = "Health";
			this.isCool = true;
			break;
		}
	}
	
	public void drawOn(Graphics2D g2) {
		g2.translate(this.getXCent(), this.getYCent());
		
		String fileName = "images/" + "Plat" + name;
		
		fileName += ".PNG";
		BufferedImage img;
		try {
			img = ImageIO.read(new File(fileName));
			g2.drawImage(img, (int) -WIDTH/2, (int) -HEIGHT/2, (int) WIDTH, (int) HEIGHT, null);
		} catch (IOException e) {
		}
		
//		g2.setColor(Color.red);
//		//g2.translate(this.getXCent(), this.getYCent());
//		g2.fill(new Rectangle2D.Double(-WIDTH/2,-HEIGHT/2,WIDTH,HEIGHT));
//		g2.setColor(Color.BLACK);
//		g2.draw(new Rectangle2D.Double(-WIDTH/2,-HEIGHT/2,WIDTH,HEIGHT));
		
		g2.translate(-this.getXCent(), -this.getYCent());
		
	}

	public double getWIDTH() {
		return WIDTH;
	}

	public void setWIDTH(double wIDTH) {
		WIDTH = wIDTH;
	}

	public double getHEIGHT() {
		return HEIGHT;
	}

	public void setHEIGHT(double hEIGHT) {
		HEIGHT = hEIGHT;
	}

	public boolean isLava() {
		return isLava;
	}

	public void setLava(boolean isLava) {
		this.isLava = isLava;
	}

	public boolean isIce() {
		// TODO Auto-generated method stub
		return isIce;
	}

	public boolean isSlime() {
		// TODO Auto-generated method stub
		return isSlime;
	}
	public boolean isCool() {
		return isCool;
	}
	public void SetCool(boolean isCool) {
		this.isCool = isCool;
	}
}

File: docs/js/Platform.js
/**
 * Platform class for static platforms with different properties
 */
class Platform extends GameObject {
    constructor(xCord, yCord, type) {
        super(xCord, yCord);
        this._isLava = false;
        this._isIce = false;
        this._isSlime = false;
        this._isCool = false;
        this.name = "";
        this.imageCache = new Map();
        
        this.setName(type);
        
        if (this._isLava) {
            this.setHeight(this.getHeight() * 0.6);
        }
    }

    setName(type) {
        switch (type) {
            case 0:
                this.name = "Truss";
                break;
            case 1:
                this.name = "Lava";
                this._isLava = true;
                break;
            case 2:
                this.name = "Ice";
                this._isIce = true;
                break;
            case 3:
                this.name = "Slime";
                this._isSlime = true;
                break;
            case 4:
                this.name = "Health";
                this._isCool = true;
                break;
        }
    }

    drawOn(ctx) {
        // Use correct file extensions based on actual files
        let fileName;
        switch (this.name) {
            case "Truss":
                fileName = "PlatTruss.PNG";
                break;
            case "Lava":
                fileName = "PlatLava.PNG";
                break;
            case "Ice":
                fileName = "PlatIce.PNG";
                break;
            case "Slime":
                fileName = "PlatSlime.png";
                break;
            case "Health":
                fileName = "PlatHealth.png";
                break;
            default:
                fileName = "Plat" + this.name + ".png";
        }
        
        let useImage = false;
        
        if (this.imageCache.has(fileName)) {
            const img = this.imageCache.get(fileName);
            
            // Check if it's a failed image marker
            if (img && img.failed) {
                // Skip drawing image, use fallback
            } else if (img && img.complete) {
                // Check if the image is actually valid
                if (img.naturalWidth && img.naturalWidth > 0) {
                    try {
                        ctx.drawImage(
                            img,
                            this.xCent - this.width / 2,
                            this.yCent - this.height / 2,
                            this.width,
                            this.height
                        );
                        useImage = true;
                    } catch (error) {
                        console.error(`ERROR: Failed to draw ${fileName}:`, error.message);
                        // Mark as failed
                        this.imageCache.set(fileName, { complete: true, failed: true });
                    }
                } else {
                    this.imageCache.set(fileName, { complete: true, failed: true });
                }
            }
        } else {
            // Load image if not cached
            const img = new Image();
            
            img.onload = () => {
                this.imageCache.set(fileName, img);
            };
            
            img.onerror = () => {
                console.error(`ERROR: Failed to load ${fileName}`);
                this.imageCache.set(fileName, { complete: true, failed: true });
            };
            
            img.src = `images/${fileName}`;
            this.imageCache.set(fileName, img);
        }
        
        // Always draw fallback rectangle if image didn't work
        if (!useImage) {
            ctx.fillStyle = this.getColorForType();
            ctx.fillRect(
                this.xCent - this.width / 2,
                this.yCent - this.height / 2,
                this.width,
                this.height
            );
            ctx.strokeStyle = 'black';
            ctx.strokeRect(
                this.xCent - this.width / 2,
                this.yCent - this.height / 2,
                this.width,
                this.height
            );
        }
    }

    getColorForType() {
        if (this._isLava) return '#ff4444';
        if (this._isIce) return '#44ddff';
        if (this._isSlime) return '#44ff44';
        if (this._isCool) return '#ffdd44';
        return '#888888'; // Default gray
    }

    isLava() {
        return this._isLava === true;
    }

    setLava(isLava) {
        this._isLava = isLava;
    }

    isIce() {
        return this._isIce === true;
    }

    isSlime() {
        return this._isSlime === true;
    }

    isCool() {
        return this._isCool === true;
    }

    setCool(isCool) {
        this._isCool = isCool;
    }
    
    // Alternative method name to match Java version
    SetCool(isCool) {
        this._isCool = isCool;
    }
} 